.evergreen/auth_oidc/setup_oidc.js

```diff
@@ -10,6 +10,16 @@ admin.runCommand({createRole: 'test1/readWrite', roles:[{role: 'readWrite', db:
 admin.runCommand({createRole: 'test2/read', roles:[{role: 'read', db: 'test'}], privileges: []});
 
 // Wait for the node to be primary.
-while (admin.runCommand('replSetGetStatus').myState != 1) {}
+var shouldQuit = false;
+function waitForReady() {
+    if (admin.runCommand('replSetGetStatus').myState != 1) {
+        setTimeout(waitForReady, 1000);
+    } else {
+        console.log("ReplicaSet is Configured!");
+        shouldQuit = true;
+    }
+}
+waitForReady();
+while (!shouldQuit) {}
```

Comment:
This is still a tight CPU loop.

Reply:
I think you can use:
```
while (admin.runCommand('replSetGetStatus').myState != 1) {
    sleep(50);
}
```

Reply:
Ah, `sleep()` is much better.  If only that were built into JS itself...